### DISCUSSION TITLE
M600 and "Move exceeds maximum extrusion" error

### DISCUSSION THREAD

**Discussion by DouglasK** [jschuh/klipper-macros#176] [Q&A] [answered]
I'm admittedly new to klipper-macros but I can happily print (without M600).

When I print a file containing an M600 code, Klipper returns "Move exceeds maximum extrusion (3.297mm^2 vs 0.640mm^2)" when I click Resume after the pause. I'm running under OctoPi if that helps. 

Some basic info:
- Printer: Ender 3 w/ direct drive + all metal hotend
- Main board on printer: SKR 1.4 Turbo
- Slicer: Cura 5.4.0, current non-beta version

Header gCode: 
```
M190 S0
M109 S0
_PRINT_START_PHASE_INIT EXTRUDER={material_print_temperature_layer_0} BED={material_bed_temperature_layer_0} NOZZLE_SIZE={machine_nozzle_size}
; Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
_PRINT_START_PHASE_PROBING
; Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).
```
Here's the Printer.cfg, renamed to .txt:
[printer.cfg.txt](https://github.com/jschuh/klipper-macros/files/12623976/printer.cfg.txt)

Here's the log file from Klipper in OctoPrint from before the print started until the error.
[plugin_klipper_debug.log](https://github.com/jschuh/klipper-macros/files/12623936/plugin_klipper_debug.log)

Here's the Gcode file, renamed to .txt. 
[PETG_sample_0.2mm_PETG_MK3S_18m.gcode.txt](https://github.com/jschuh/klipper-macros/files/12623953/PETG_sample_0.2mm_PETG_MK3S_18m.gcode.txt)

I've been working on this for a while, but am hitting my head against a wall. Any help is appreciated, including RTFM (provided one points me to the correct FM!).

Thanks for any help.

**Comment by jschuh** [comment, marked as answer]
It looks like Klipper is choking on the tip forming moves during unload. I'd suggest increasing `max_extrude_cross_section` in your Klipper config to something like `10`. The default for this is nozzle_diameter^2 * 4, which is honestly pretty easy to exceed in a variety of normal printing moves.

**Comment by lmarcondes** [comment]
This was the only discussion topic I could find mentioning the `Move exceeds maximum extrusion` error, which I've been having as well. I traced my issue to the `_PRINT_START_PHASE_PURGE` macro, specifically to the changes made in 2d77d095c69c77ca6d5cd919fe3ef870ef61a29a. Without these additions I was able to avoid the issue altogether.

![image](https://github.com/jschuh/klipper-macros/assets/33609736/f596bcd4-26ae-4d4a-b41b-cc5a1cd15618)

**Reply by ku8ec** [reply]
Same for me. I was getting that error too and rolling back the [2d77d09](https://github.com/jschuh/klipper-macros/commit/2d77d095c69c77ca6d5cd919fe3ef870ef61a29a) edits helped.

**Reply by Twinki14** [reply]
Having this exact same issue, and the fix above ([reverting 2d77d09](https://github.com/jschuh/klipper-macros/commit/2d77d095c69c77ca6d5cd919fe3ef870ef61a29a)) worked for me as well.

@jschuh This seems like a potential bug?
